### PR TITLE
Enable Strided Tensor for Training Only

### PR DESCRIPTION
**Pull request by iK1D**
Only enable strided tensor to traning, so we can keep the binary size as remain for inference.

### CODE REVIEW

File: onnxruntime/core/framework/execution_frame.h [hunk 1]
```diff
@@ -199,8 +199,7 @@ class ExecutionFrame final : public IExecutionFrame {
                                                   bool create_fence);
 
   Status AllocateTensorWithPreAllocateBufferHelper(OrtValue& ort_value, void* pBuffer, MLDataType element_type,
```
Comment (souptc) [inline onnxruntime/core/framework/execution_frame.h:201]: you don't need stride for training build?

Comment (iK1D) [reply onnxruntime/core/framework/execution_frame.h:201]: The call passes in its own shape, which may be different from original shape, so using the original strides might be long. Ideally we need to add strides to Output() call and all following functions to take new strides. For now, I am thinking we require user to set the strides accordingly after the Output() call, or do you think it's still better we add the parameter for all the functions? Note that we need the #ENABLE_TRAINING, which will make the code complex.

**Review by souptc — APPROVED** [review]

**Review by skottmckay — APPROVED** [review]

**Comment by skottmckay** [comment]
Merge with latest master should fix the TVM build error.

**Comment by chilo-ms** [comment]
This PR is already in rel-1.11.0 branch, so remove the label.

**Pull request merged**